### PR TITLE
HPCC-23587 eclccserver on demand

### PR DESCRIPTION
**Pull request by richardkchapman**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [x] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
Manual in minikube

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review

File: dockerfiles/eclccserver/Dockerfile [hunk 1]
```diff
@@ -20,6 +20,9 @@
 ARG BUILD_LABEL
 FROM hpccsystems/platform-core:${BUILD_LABEL}
 
+USER root
```
Comment (richardkchapman) [inline dockerfiles/eclccserver/Dockerfile]: Possibly should be in a base image

Comment (richardkchapman) [reply dockerfiles/eclccserver/Dockerfile]: Also may need to make sure RBAC is properly set up to allow eclccserver to do this - it's working for me but I may have changed the default access controls. I'll raise a Jira to check that.

Comment (richardkchapman) [reply dockerfiles/eclccserver/Dockerfile]: Also may need to make sure RBAC is properly set up to allow eclccserver to do this - it's working for me but I may have changed the default access controls. I'll raise a Jira to check that.

File: ecl/eclccserver/eclccserver.cpp [hunk 2]
```diff
@@ -452,6 +452,46 @@ class EclccCompileThread : implements IPooledThread, implements IErrorReporter,
...
+            jobYaml.replaceString("%args", args.str());
+            StringBuffer output, error;
+            DBGLOG("Using job yaml %s", jobYaml.str());
+            unsigned ret = runExternalCommand(output, error, "kubectl apply -f -", jobYaml.str());
+            DBGLOG("kubectl output: %s", output.str());
+            if (error.length())
+                DBGLOG("kubectl error: %s", error.str());
+            if (ret)
+            {
+                DBGLOG("Using job yaml %s", jobYaml.str());
```
Comment (ghalliday) [inline ecl/eclccserver/eclccserver.cpp]: Already traced on line 466.  I suspect one of them should be deleted.

File: system/jlib/jfile.cpp [hunk 3]
```diff
@@ -6756,6 +6756,7 @@ extern jlib_decl void removeSentinelFile(IFile * sentinelFile)
 
 extern jlib_decl void writeSentinelFile(IFile * sentinelFile)
 {
+#ifndef _CONTAINERIZED
```
Comment (ghalliday) [inline system/jlib/jfile.cpp]: Should also comment out the code in removeSentinalFile()

**Review by ghalliday — CHANGES_REQUESTED** [review]
@richardkchapman a couple of minor issues

**Review by ghalliday — COMMENTED** [review]
A couple of trivial comments

File: dockerfiles/hpcc/values.yaml [hunk 4]
```diff
@@ -33,11 +33,11 @@ esp:
 - name: myesp
   replicas: 1
 
-notroxie:
+xroxie:
```
[the remark this answers is not in the document]
Comment (richardkchapman) [reply dockerfiles/hpcc/values.yaml]: I treat values.yaml as a bit of a scratchpad for testing the bit currently under development.

Yes, at some point we should start regarding it as a proper example file, with documentation inline (comments) for all settings. There is a Jira open to do so.

**Review by ghalliday — APPROVED** [review]

**Pull request merged**